eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java

```diff
@@ -50,30 +50,28 @@ public void beginTree(DetailAST rootAST) {
     @Override
     public int[] getRequiredTokens() {
         return new int[] {
-            TokenTypes.PACKAGE_DEF,
             TokenTypes.IMPORT,
             TokenTypes.METHOD_DEF
         };
     }
 
     @Override
     public void visitToken(DetailAST token) {
-        if (isImplPackage) {
-            return;
-        }
-
         switch (token.getType()) {
-            case TokenTypes.PACKAGE_DEF:
-                String packageName = FullIdent.createFullIdent(token.findFirstToken(TokenTypes.DOT)).getText();
-                isImplPackage = packageName.contains(".implementation");
-                break;
             case TokenTypes.IMPORT:
                 // Add all imported classes into a map, key is the name of class and value is the full package path of class.
                 final String importClassPath = FullIdent.createFullIdentBelow(token).getText();
                 final String className = importClassPath.substring(importClassPath.lastIndexOf(".") + 1);
                 simpleClassNameToQualifiedNameMap.put(className, importClassPath);
                 break;
+            case TokenTypes.CLASS_DEF:
+                final DetailAST modifiersToken = token.findFirstToken(TokenTypes.MODIFIERS);
```

Comment:
There's a CheckUtil method you used in the same file that gives you the modifier.
`final AccessModifier accessModifier = CheckUtil.getAccessModifierFromModifiersToken(modifiersToken);`